### PR TITLE
Add junit test for UnusedAlgorithmFinder

### PR DESCRIPTION
**Pull request by dmytro-taranov**
Refs [#25730](https://github.com/apache/shardingsphere/issues/25730)

Changes proposed in this pull request:
-  Add JUnit test for UnusedAlgorithmFinder.find(..)

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    private static final String USED_DATABASE_SHARDING_ALGORITHM = "used_database_sharding_algorithm";
+    
+    private static final String USED_DATABASE_SHARDING_DEFAULT_ALGORITHM = "used_database_sharding_default_algorithm";
+    
+    private static final String UNUSED_ALGORITHM = "unused_algorithm";
+    
+    @Test
+    void assertFind() {
+        ShardingRuleConfiguration ruleConfig = new ShardingRuleConfiguration();
+
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:52]: The empty lines in method could be removed for clean code

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    
+    private static final String USED_DATABASE_SHARDING_DEFAULT_ALGORITHM = "used_database_sharding_default_algorithm";
+    
+    private static final String UNUSED_ALGORITHM = "unused_algorithm";
+    
+    @Test
+    void assertFind() {
+        ShardingRuleConfiguration ruleConfig = new ShardingRuleConfiguration();
+        
+        ShardingTableRuleConfiguration shardingTableRuleConfiguration = getShardingTableRuleConfiguration();
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:53]: `shardingTableRuleConfiguration` name could be `shardingTableRuleConfig` for less code

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        Map<String, AlgorithmConfiguration> allAlgorithms = getAlgorithms();
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:54]: `allAlgorithms` could be removed and embed `getAlgorithms()` in `putAll`

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        
+        ruleConfig.getTables().add(shardingTableRuleConfiguration);
+        ruleConfig.getShardingAlgorithms().putAll(allAlgorithms);
+        ruleConfig.setDefaultDatabaseShardingStrategy(new StandardShardingStrategyConfiguration("order_id", USED_DATABASE_SHARDING_DEFAULT_ALGORITHM));
+        ruleConfig.setDefaultTableShardingStrategy(new StandardShardingStrategyConfiguration("order_id", USED_TABLE_SHARDING_DEFAULT_ALGORITHM));
+        
+        Collection<String> unusedAlgorithmsCollection = UnusedAlgorithmFinder.find(ruleConfig);
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:61]: `unusedAlgorithmsCollection` name could be `actual`

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        assertThat(unusedAlgorithmsCollection.size(), is(1));
+        assertTrue(unusedAlgorithmsCollection.contains(UNUSED_ALGORITHM));
+    }
+    
+    private ShardingTableRuleConfiguration getShardingTableRuleConfiguration() {
+        ShardingTableRuleConfiguration shardingTableRuleConfiguration = new ShardingTableRuleConfiguration("t_order", null);
+        shardingTableRuleConfiguration.setTableShardingStrategy(new StandardShardingStrategyConfiguration("order_id", USED_TABLE_SHARDING_ALGORITHM));
+        shardingTableRuleConfiguration.setDatabaseShardingStrategy(new StandardShardingStrategyConfiguration("order_id", USED_DATABASE_SHARDING_ALGORITHM));
+        
+        return shardingTableRuleConfiguration;
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:73]: `shardingTableRuleConfiguration` name could be `result`

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import org.apache.shardingsphere.infra.config.algorithm.AlgorithmConfiguration;
+import org.apache.shardingsphere.sharding.api.config.ShardingRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.rule.ShardingTableRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.sharding.StandardShardingStrategyConfiguration;
+import org.apache.shardingsphere.sharding.distsql.handler.update.UnusedAlgorithmFinder;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Properties;
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:30]: You could import `src/resources/code-style-idea.xml` in IDEA and format importing

**Review by sandynz — CHANGES_REQUESTED** [review]

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java (hunk 3 above)
Comment (dmytro-taranov) [reply features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java]: Okay, thanks for the hint. Changed

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java (hunk 6 above)
Comment (dmytro-taranov) [reply features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:30]: @sandynz I imported this file and formatted the code before the first commit. Unfortunately, after second reformatting the situation with the imports did not change. I also checked the imports in other classes, and the logic of their arrangement is the same as mine. Could you please point out what exactly is wrong with imports? I'd appreciate it, and I'll fix them.

**Comment by dmytro-taranov** [comment]
Hi. @sandynz. Thanks for the review. And thanks for the hint. I will double-check the code of conduct

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java (hunk 6 above)
Comment (sandynz) [reply features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/UnusedAlgorithmFinderTest.java:30]: Sorry, that's my mistake. I tested on my local machine, that's OK.

**Comment by sandynz** [comment]
> Hi. @sandynz. Thanks for the review. And thanks for the hint. I will double-check the code of conduct

Nice. Looking forward to more contributions.

**Pull request merged**